File: FE-RESTAURANT/src/interfaces/BranchResponseDTO.ts
import { type BranchStatus } from './enums/BranchStatus';
// Updated interface with isInUse field

export interface BranchResponseDTO {
  id: string;
  name: string;
  address: string;
  phone: string;
  description: string;
  status: BranchStatus;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
  isActive: boolean;
  isInUse?: boolean; // Chi nhánh có đang được sử dụng không
}

File: RESTAURANT/src/main/java/com/poly/restaurant/dtos/BranchResponseDTO.java
package com.poly.restaurant.dtos;

import com.poly.restaurant.entities.enums.BranchStatus;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BranchResponseDTO {
    private Long id;
    private String name;
    private String address;
    private String province;
    private String district;
    private String ward;
    private String phone;
    private String description;
    private BranchStatus status;
    private String createdBy;
    private Double latitude;
    private Double longitude;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean isInUse; // Chi nhánh có đang được sử dụng không

}